src/main/java/carpet/settings/ParsedRule.java

```diff
@@ -156,7 +180,7 @@ else if (type.isEnum())
             if (!value.equals(get()) || source == null)
             {
                 this.field.set(null, value);
-                if (source != null) CarpetServer.settingsManager.notifyRuleChanged(source, this, stringValue);
+                if (source != null) settingsManager.notifyRuleChanged(source, this, stringValue);
```

Comment:
not so sure about this one - this means that extensions will miss rule updates form non-`carpet` managers. Not sure if this is good, since the settings manager can be fetched per rule via 'this' send as a param

Reply:
Well, I'd say when most extensions register their rule observers they usually only want to get changes from their rules. Anyway, extensions can still register observers for other managers if later than extensions load they add their observers by checking for extensions at `CarpetServer.extensions`.

Because also else that would make each manager having their own observer list rather pointless, since those would never be called.

Reply:
I've checked in every extension [here](https://github.com/gnembon/fabric-carpet/wiki/List-of-Carpet-extensions) and all (the only) rule observers I found only care about their rule changes.

Reply:
If you really want, maybe having 2 observer types? (global/per manager)
Or making observers static among all SettingsManager?

Reply:
I get that having it 'carpet' instance specific may not be intuitive with custom managers, but I get the appeal of one global one. Lets do what you suggested - having a static one shared between managers, and for each manager. Each manager would notify all static observers as well as their own. This way you can subscribe to what you want